### ISSUE TITLE
No Screensharing on Zoom

### ISSUE THREAD

**Issue by authenticastrology** [pop-os/cosmic-epoch#802]
Although I can't seem to even open Zoom at the moment, a few days ago when I could, I could not get screensharing to work.

**Comment by authenticastrology** [comment]
I ran journalctl after trying to screenshare. I am sure some of this is irrelevant (noob). But thought the relevant parts might help in fixing the issue.

Aug 26 12:23:11 pop-os xdg-desktop-por[1840]: A backend call failed: No such interface “org.freedesktop.impl.portal.RemoteDesktop” on object at path /org/freedesktop/portal/desktop
Aug 26 12:23:11 pop-os xdg-desktop-por[1840]: Failed to close session implementation: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Object does not exist at path “/org/freedesktop/portal/desktop/session/1_115/zoomcast1”
Aug 26 12:23:18 pop-os systemd[1581]: app-cosmic-com.system76.CosmicAppList-3156.scope: Consumed 29.722s CPU time.
Aug 26 12:23:22 pop-os systemd[1581]: app-cosmic-com.system76.CosmicAppList-3903.scope: Couldn't move process 3903 to requested cgroup '/user.slice/user-1000.slice/user@1000.service/app.slice/app-cosmic-com.system76.CosmicAppList-3903.scope': No such process

**Comment by authenticastrology** [comment]
Perhaps it's assuming, since it is an Ubuntu version of Zoom, that Gnome is being used and so it is looking for Gnome and not finding it?

**Comment by Zie0** [comment]
This issue appears related to https://github.com/pop-os/cosmic-epoch/issues/736

**Comment by alancleary** [comment]
@authenticastrology Some additional steps may be required to enable screen sharing on Linux. These are documented on the Arch Linux Wiki: https://wiki.archlinux.org/title/Zoom_Meetings#Screen_share

**Comment by authenticastrology** [comment]
Yes, it is easy to make mistakes. I did look to see if there was something
reported. But sometimes we all miss things (my excuse and I am sticking
with it...lol.)

**Comment by authenticastrology** [comment]
Thank you Alan, I will give that a look.

**Comment by authenticastrology** [comment]
Alan, 
Any idea of where I would need to go to set that environment variable? Haven't had any luck. Wayland share is already enabled.

**Comment by alancleary** [comment]
> Any idea of where I would need to go to set that environment variable? Haven't had any luck. Wayland share is already enabled.

There's a few different ways you can do it depending on if you want the variable to be set for just your user or globally.<sup>[1](https://wiki.archlinux.org/title/Environment_variables#Defining_variables)</sup> In this scenario I would personally set it globally by adding it to your `/etc/environment` file. For the change to take effect you'll need to either reboot the system or restart your session.

**Comment by authenticastrology** [comment]
I have tried pretty much everything that was suggested and screensharing simply doesn't work using the app with Cosmic. Doing a meeting from a browser misses some needed functionality. Anyone else with this issue? Hope it gets fixed soon. For now I have dual boot set up with Ubuntu for when I need to host meetings. This is probably the only thing I cannot get to work with Cosmic. Other than that, even being in Alpha stage, it is pretty perfect.

**Comment by gmjain** [comment]
I'm facing the exact same issue. As soon as zoom screen sharing starts working, I'll move to cosmic to use as my daily driver. I'm willing to help debug this issue as well.

**Comment by alancleary** [comment]
Based on [this Zoom Community thread](https://community.zoom.com/t5/Zoom-Meetings/share-screen-linux-wayland-broken/m-p/184192), it sounds like Zoom is the issue here, i.e. they only develop against the Gnome DE on Ubuntu. So it's doubtful that the COSMIC team will be able to come up with a robust solution.

Fortunately the Zoom team has [posted a workaround](https://support.zoom.com/hc/en/article?id=zm_kb&sysparm_article=KB0059085) that I have confirmed works with COSMIC. Here are the instructions copy-pasted from the linked article:

## Use a third-party virtual camera

**Note**: This workaround uses a third-party virtual camera tool that is not officially supported by Zoom. Refer to [OBS Support](https://obsproject.com/help) if you have any issues.

1. [Install OBS](https://obsproject.com/wiki/install-instructions#linux) on your Linux distro. Make sure your system meets all prerequisites for OBS.
2. Open OBS and [create a scene using Display Capture (share the entire screen) or Window Capture (share a specific window)](https://obsproject.com/wiki/OBS-Studio-Overview#scenes-and-sources).
3. In the **Controls** panel, click **Start Virtual Camera**.
4. [Start](https://support.zoom.com/hc/en/article?id=zm_kb&sysparm_article=KB0061821) or [join](https://support.zoom.com/hc/en/article?id=zm_kb&sysparm_article=KB0060732) a Zoom meeting.
5. Share your screen and select **Content from a 2nd Camera**.
6. In the top-right corner, click **Switch Camera** until the OBS virtual camera is displayed as the screen share.

**Comment by alancleary** [comment]
I'm not sure if it's the recent COSMIC update (epoch-1.0.0-alpha.3) or the recent Zoom update (6.2.10) but screen sharing seems to be working now (at least on Arch). I'll note, however, that I had to manually set the screen capture mode in the Zoom settings for it to work: **Settings > Share Screen > Advanced > Screen capture mode on Wayland = Pipewire Mode.** I'm going to keep the OBS workaround I previously described ready to go since historically Zoom seems prone to breaking...

**Comment by matdave** [comment]
I can confirm that this seems to be working on the latest version of Zoom.

**Comment by authenticastrology** [comment]
I can't get it to work on the Pop version of Cosmic and now screensharing is broken in the browser as well.

**Comment by alancleary** [comment]
@authenticastrology Have you tried the workaround [I previously described](https://github.com/pop-os/cosmic-epoch/issues/802#issuecomment-2430435930)? Screen sharing stopped working in my Zoom again yesterday so I had to use the work around...

**Comment by authenticastrology** [comment]
Alan, I really need Window Capture on OBS and not screen capture. OBS doesn't seem to have window capture anymore for some reason. Do you know how to access an older version of OBS?
The issue I am running into is that I counsel and need to see the faces of my clients. Doing so with just the virtual camera and Screen capture creates the Infinity Mirror issue.

**Comment by authenticastrology** [comment]
Though I was actually able to get Window capture on OBS to work (it is an option under screen capture now), after doing so with the latest updates from cosmic screenshare does not appear as an option in OBS either flatpak or system versions. I noticed the latest update contained a screeencasting change. I imagine that is at the root of it. I will just stay back with the version of Cosmic I have from a few days ago until hopefully they provide a fix.

**Comment by ntnsndr** [comment]
Bumping this issue, as it is persisting. I'm using Flatpak Zoom on PopOS COSMIC and when I try to share a screen, the dialog just goes away with no screen sharing.

@alancleary's suggestion of Pipewire Mode doesn't help. Nor does using OBS; the virtual camera doesn't appear to be starting up from there. (I don't have that button in Controls but it is an option in the panel applet.)

Looks like on Reddit people are reverting to X11 gnome because of this: https://www.reddit.com/r/pop_os/comments/1izsons/zoom_screen_sharing/

Is there any hope of a fix, or is the an endemic Wayland issue?

**Comment by matdave** [comment]
@ntnsndr if the flatpak is giving you issues and you use chrome or a chromium based browser you can go to https://pwa.zoom.us/wc and install that as a PWA

**Comment by deadeyejoe** [comment]
> Bumping this issue, as it is persisting. I'm using Flatpak Zoom on PopOS COSMIC and when I try to share a screen, the dialog just goes away with no screen sharing.

I've run into the exact same problem as @ntnsndr, running `ps aux | grep portal` showed that there were two competing portal backends running; `xdg-desktop-portal-gtk` and `xdg-desktop-portal-cosmic`. Purging the gtk package using `sudo apt purge xdg-desktop-portal-gtk` fixed the problem.

**Comment by wowitsjack** [comment]
> > Bumping this issue, as it is persisting. I'm using Flatpak Zoom on PopOS COSMIC and when I try to share a screen, the dialog just goes away with no screen sharing.
> 
> I've run into the exact same problem as [@ntnsndr](https://github.com/ntnsndr), running `ps aux | grep portal` showed that there were two competing portal backends running; `xdg-desktop-portal-gtk` and `xdg-desktop-portal-cosmic`. Purging the gtk package using `sudo apt purge xdg-desktop-portal-gtk` fixed the problem.

This is the fix, additionally I had to manually start the COSMIC portal. Then Discord and Zoom worked.

**Comment by ntnsndr** [comment]
Can we confirm there are no potential risks from removing that? Is it definitely vestigial?

**Comment by programming-for-people** [comment]
I was thrilled to try this (removing the portal-gtk), and then... my message summary window in Betterbird went blank without xdg-desktop-portal-gtk.  So it seems some apps need it.

it's in the default configuration for cosmic...

usr/share/xdg-desktop-portal/cosmic-portals.conf contains:

default=cosmic;gtk;

**Comment by nitin88** [comment]
This issue fixed in the latest version of Zoom (v7.0.5).
